backend/server.js

```diff
@@ -0,0 +1,224 @@
+const express = require("express");
+const cors = require("cors");
+const multer = require("multer");
+
+const upload = multer();
+
+const cookieParser = require("cookie-parser");
+const app = express();
+
+const MongoClient = require("mongodb").MongoClient;
+
+const passwordCheck = require(__dirname + "/utilities/passwordCheck.js");
+const generateRandomString = require(__dirname +
+  "/utilities/generateRandomString.js");
+const passwordHash = require("password-hash");
+
+// app.listen(4000, () => {
+//    // LOCAL SERVER
+//    console.log("Running on port 4000");
+// });
+//_____________________MOGODB Setup___________________
+
+let dataBaseUrl =
+  "mongodb+srv://admin:WhqtQsNqxt@cluster0-58fms.mongodb.net/test?retryWrites=true&w=majority";
```

Comment:
ahh, so `WhqtQsNqxt` appears to be the password for your cloud db connection! You very much don't want this on github 😅bad actors scrape github looking for stuff like this. It's not a huge deal since I'm guessing you don't have any sensitive data in there, but you'll want to update this (and then change your password).

There are a bunch of solutions for "secrets", stuff like passwords and API keys that the app needs, but that you don't want to check into Git. The simplest is to create a `secrets.js` file:

```
module.exports.databasePassword = 'abc123';
```

Then, add it to your .gitignore:

```
// .gitignore
secrets.js
```

Then you can require it and use it in `server.js`:

```
const {databasePassword} = require('./secrets.js');
const databaseUri = `mongodb+src://admin:${databasePassword}@ cluster0-58fms.mongodb.net/test?retryWrites=true&w=majority`
```

(as a sidenote: this is more-commonly called a `uri`, not a `url`. I'm not 100% sure why.)